packages/components/src/number-control/index.tsx

```diff
@@ -60,42 +59,24 @@ function UnforwardedNumberControl(
 		spinControls = 'none';
 	}
 
+	if ( typeof valueProp === 'number' ) {
+		// TODO: deprecate `value` as a `number`
```

Comment:
To be implemented in a later PR, once all usages of `NumberControl` are migrated to using a `string` for the `value` prop

Reply:
We decided to use `string` as the type for the `value` prop for a few reasons:
- we wanted to have one recommended way to pass `value`, since the ambiguity of this type is creeping across many higher level components (we're noticing this especially as we try to make progress with the TypeScript migration)
- this type aligns closer with native HTML input elements
- using string would also align with the convention according to which controlled components pass `''` to represent an empty value (while `undefined` denotes the uncontrolled version of the component).

We won't drop support for `number` in order not to introduce a breaking change, but the plan is to discourage it by marking it as a deprecated type for the prop